### DISCUSSION TITLE
MX26L6420-90

### DISCUSSION THREAD

**Discussion by ramonxxl** [sanni/cartreader#373] [Need help]
hello I have a pcb with 29l3211, on it I can record directly from my pcb, but I'm trying something new with the 26l6420, pinouts that were changed, we and A21 that was added, and the same is not finding in the recorder, ID Type 1: C2C2
ID Type 2: C2FC
Unknow flash, Can anyone help with this matter please.

images

[https://imgur.com/gallery/UmePiWa](url)

**Comment by sanni** [comment]
Try replacing FLASH.ino with this changed version, upload to the Arduino and then see how far it gets.

[FLASH.zip](https://github.com/sanni/cartreader/files/8808469/FLASH.zip)

**Reply by ramonxxl** [reply]
I will test it, thanks sanni

**Comment by ramonxxl** [comment]
Hello Sanni, now the recognition of the 26L6420 flash worked,
delete = ok
blank check = ok
write = ?
recording takes more than 20 minutes and doesn't record anything, it seems like the screen freezes

**Comment by sanni** [comment]
Are you sure your SNES PCB works with 26L6420? 

The 29l3211 has an 8bit mode that turns the data pin Q15 into the first address pin (A -1, address negative one).
The 26L6420 does not seem to have such an 8bit mode so Q15 always is Q15 and therefore SNES A0 is not connected to an address line on the flashrom and all following address lines are also connected wrong as a result.
I'm pretty sure the 26L6420 needs a specific 26L6420 SNES PCB and can't be used with the 29l3211 8bit PCB. 
They only behave the same when both chips are used in 16bit mode.

But ofc I could be wrong since I have no hands-on experience with the 26L6420.

**Comment by ramonxxl** [comment]
Sanni, thank you for answering, so my PCB I'm testing for 26L6420, at first it only serves for 29l3211, so I want to test for 26L6420, as soon as it works I will fix the schematic to be able to produce the official PCB for 26L. I thought I just needed to change the a21 and WE.
